File: test-suite/java/guards_buggy.java
package com.example.inventory;

class OrderTotal {

    int catalogTotal(Customer customer) {
        return customer.profile.stats.orders.total;
    }

    String shipmentLabel(ShipClient client) {
        return client.conn.region.host.name;
    }

    int safeTotal(Customer customer) {
        // ternary without a null test is not a guard
        int fallback = 0;
        boolean eager = customer.profile.stats.orders.total > fallback;
        return eager ? customer.profile.stats.orders.total : fallback;
    }
}

File: test-suite/csharp/guards_buggy.cs
namespace Inventory.Telemetry;

class Telemetry
{
    public int TimeoutMs(Config cfg)
    {
        return cfg.Http.Stack.Retry.TimeoutMs;
    }

    public string Region(ShipClient client)
    {
        return client.Conn.Region.Host.Name;
    }

    public int RawFallback(Config cfg)
    {
        // ?? coalescing alone never checks the chain it reads
        return cfg.Http.Stack.Retry.TimeoutMs + (cfg.RetryLimit ?? 0);
    }
}
